[CI] mixed-cluster:v6.0.0#mixedClusterTestRunner failure

**Build scan**:
https://gradle-enterprise.elastic.co/s/64inre6ljqgcc

**Repro line**:
```
./gradlew ':qa:mixed-cluster:v6.0.0#mixedClusterTestRunner' -Dtests.seed=4925E97ABC52AD2B -Dtests.class=org.elasticsearch.backwards.MixedClusterClientYamlTestSuiteIT -Dtests.method="test {p0=nodes.stats/20_response_filtering/Nodes Stats filtered using both includes and excludes filters}" -Dtests.security.manager=true -Dtests.locale=sr -Dtests.timezone=America/Punta_Arenas -Dcompiler.java=11 -Druntime.java=8
```

**Reproduces locally?**:
Haven't tried

**Applicable branches**:
6.8

**Failure history**:
https://gradle-enterprise.elastic.co/scans/failures?failures.failureClassification=all_failures&failures.failureMessage=Execution%20failed%20for%20task%20*%0A%3E%20Process%20%27kill%20%5B-9,%20*%20finished%20with%20non-zero%20exit%20value%201&search.relativeStartTime=P7D&search.timeZoneId=Europe/Athens

**Failure excerpt**:
```
java.lang.AssertionError: expected total memory to be positive, got: -1
	at org.elasticsearch.monitor.os.OsStats$Mem.<init>(OsStats.java:267) ~[elasticsearch-6.8.17-SNAPSHOT.jar:6.8.17-SNAPSHOT]
	at org.elasticsearch.monitor.os.OsStats.<init>(OsStats.java:55) ~[elasticsearch-6.8.17-SNAPSHOT.jar:6.8.17-SNAPSHOT]
	at org.elasticsearch.common.io.stream.StreamInput.readOptionalWriteable(StreamInput.java:777) ~[elasticsearch-6.8.17-SNAPSHOT.jar:6.8.17-SNAPSHOT]
	at org.elasticsearch.action.admin.cluster.node.stats.NodeStats.readFrom(NodeStats.java:227) ~[elasticsearch-6.8.17-SNAPSHOT.jar:6.8.17-SNAPSHOT]
	at org.elasticsearch.action.support.nodes.TransportNodesAction$AsyncAction$1.read(TransportNodesAction.java:199) ~[elasticsearch-6.8.17-SNAPSHOT.jar:6.8.17-SNAPSHOT]
	at org.elasticsearch.action.support.nodes.TransportNodesAction$AsyncAction$1.read(TransportNodesAction.java:195) ~[elasticsearch-6.8.17-SNAPSHOT.jar:6.8.17-SNAPSHOT]
	at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.read(TransportService.java:1107) ~[elasticsearch-6.8.17-SNAPSHOT.jar:6.8.17-SNAPSHOT]
	at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.read(TransportService.java:1094) ~[elasticsearch-6.8.17-SNAPSHOT.jar:6.8.17-SNAPSHOT]
	at org.elasticsearch.transport.TcpTransport.handleResponse(TcpTransport.java:970) ~[elasticsearch-6.8.17-SNAPSHOT.jar:6.8.17-SNAPSHOT]
	at org.elasticsearch.transport.TcpTransport.messageReceived(TcpTransport.java:952) ~[elasticsearch-6.8.17-SNAPSHOT.jar:6.8.17-SNAPSHOT]
	at org.elasticsearch.transport.TcpTransport.inboundMessage(TcpTransport.java:763) ~[elasticsearch-6.8.17-SNAPSHOT.jar:6.8.17-SNAPSHOT]
	at org.elasticsearch.transport.netty4.Netty4MessageChannelHandler.channelRead(Netty4MessageChannelHandler.java:53) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:323) ~[?:?]
	at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:297) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) ~[?:?]
	at io.netty.handler.logging.LoggingHandler.channelRead(LoggingHandler.java:241) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:340) ~[?:?]
	at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1434) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:362) ~[?:?]
	at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:348) ~[?:?]
	at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:965) ~[?:?]
	at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:163) ~[?:?]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:656) ~[?:?]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysPlain(NioEventLoop.java:556) ~[?:?]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:510) ~[?:?]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:470) ~[?:?]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:909) ~[?:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_291]
```


Could be related to the JAVA related updates:
https://github.com/elastic/elasticsearch/commit/14f3273dd2d3cf5f4d00d7cec601caca00701d71
https://github.com/elastic/elasticsearch/commit/bb580a45baebca48ac759d6d18499b0e70f12d99

--- a/server/src/main/java/org/elasticsearch/monitor/os/OsStats.java
+++ b/server/src/main/java/org/elasticsearch/monitor/os/OsStats.java
@@ -193,6 +193,8 @@ public class OsStats implements Writeable, ToXContentFragment {
             } else {
                 // If we have a node in the cluster without the bug fix for
                 // negative memory values, we need to coerce negative values to 0 here.
+                // The relevant bug fix was added for 7.8.0 in https://github.com/elastic/elasticsearch/pull/57317
+                // and for 6.8.14 in https://github.com/elastic/elasticsearch/pull/68554
                 this.total = Math.max(0, in.readLong());
                 this.free = Math.max(0, in.readLong());
             }
@@ -263,6 +265,7 @@ public class OsStats implements Writeable, ToXContentFragment {
             } else {
                 // If we have a node in the cluster without the bug fix for
                 // negative memory values, we need to coerce negative values to 0 here.
+                // The relevant bug fix was added for 7.2.0 and 6.8.2 in https://github.com/elastic/elasticsearch/pull/42725
                 this.total = Math.max(0, in.readLong());
                 this.free = Math.max(0, in.readLong());
             }

--- a/server/src/main/java/org/elasticsearch/monitor/os/OsStats.java
+++ b/server/src/main/java/org/elasticsearch/monitor/os/OsStats.java
@@ -185,10 +185,17 @@ public class OsStats implements Writeable, ToXContentFragment {
         }
 
         public Swap(StreamInput in) throws IOException {
-            this.total = in.readLong();
-            assert total >= 0 : "expected total swap to be positive, got: " + total;
-            this.free = in.readLong();
-            assert free >= 0 : "expected free swap to be positive, got: " + total;
+            if (in.getVersion().onOrAfter(Version.V_6_8_14) || in.getVersion().onOrAfter(Version.V_7_8_0)) {
+                this.total = in.readLong();
+                assert total >= 0 : "expected total swap to be positive, got: " + total;
+                this.free = in.readLong();
+                assert free >= 0 : "expected free swap to be positive, got: " + total;
+            } else {
+                // If we have a node in the cluster without the bug fix for
+                // negative memory values, we need to coerce negative values to 0 here.
+                this.total = Math.max(0, in.readLong());
+                this.free = Math.max(0, in.readLong());
+            }
         }
 
         @Override
@@ -248,10 +255,17 @@ public class OsStats implements Writeable, ToXContentFragment {
         }
 
         public Mem(StreamInput in) throws IOException {
-            this.total = in.readLong();
-            assert total >= 0 : "expected total memory to be positive, got: " + total;
-            this.free = in.readLong();
-            assert free >= 0 : "expected free memory to be positive, got: " + total;
+            if (in.getVersion().onOrAfter(Version.V_6_8_2) || in.getVersion().onOrAfter(Version.V_7_2_0)) {
+                this.total = in.readLong();
+                assert total >= 0 : "expected total memory to be positive, got: " + total;
+                this.free = in.readLong();
+                assert free >= 0 : "expected free memory to be positive, got: " + total;
+            } else {
+                // If we have a node in the cluster without the bug fix for
+                // negative memory values, we need to coerce negative values to 0 here.
+                this.total = Math.max(0, in.readLong());
+                this.free = Math.max(0, in.readLong());
+            }
         }
 
         @Override

